pkg/schemacmp/table.go

```diff
@@ -235,6 +235,168 @@ func (indexMap) JoinWithNil(value Lattice) (Lattice, error) {
 	return nil, nil
 }
 
+type partitionDefinition struct {
+	lessThan []string
+	inValues [][]string
+}
+
+func (a partitionDefinition) Equals(other Equality) bool {
+	b, ok := other.(partitionDefinition)
+	if !ok {
+		return false
+	}
+	if len(a.lessThan) != len(b.lessThan) {
+		return false
+	}
+	for i := 0; i < len(a.lessThan); i++ {
+		if a.lessThan[i] != b.lessThan[i] {
+			return false
+		}
+	}
+	if len(a.inValues) != len(b.inValues) {
+		return false
+	}
+	for i := 0; i < len(a.inValues); i++ {
+		av := a.inValues[i]
+		bv := b.inValues[i]
+		if len(av) != len(bv) {
+			return false
+		}
+		for j := 0; j < len(av); j++ {
+			if av[j] != bv[j] {
+				return false
+			}
+		}
+	}
+	return true
+}
+
+// encodePartitionDefinitionToLattice collects the necessary information for comparing a partition.
+func encodePartitionDefinitionToLattice(pd model.PartitionDefinition) Lattice {
+	return EqualitySingleton(partitionDefinition{pd.LessThan, pd.InValues})
+}
+
+type partitionMap map[string]Lattice
+
+func (partitionMap) New() LatticeMap {
+	return make(partitionMap)
+}
+
+func (a partitionMap) Get(key string) Lattice {
+	val, ok := a[key]
+	if !ok {
+		return nil
+	}
+	return val
+}
+
+func (a partitionMap) Insert(key string, value Lattice) {
+	a[key] = value
+}
+
+func (a partitionMap) ForEach(f func(key string, value Lattice) error) error {
+	for key, value := range a {
+		if err := f(key, value); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (partitionMap) CompareWithNil(value Lattice) (int, error) {
+	return 1, nil
+}
+
+func (partitionMap) ShouldDeleteIncompatibleJoin() bool {
+	return false
+}
+
+func (partitionMap) JoinWithNil(value Lattice) (Lattice, error) {
+	return value, nil
+}
+
+const (
+	partitionTupleIndexType = iota
+	partitionTupleIndexExpr
+	partitionTupleIndexColumns
+	partitionTupleIndexEnable
+	partitionTupleIndexDefinitions
+	partitionTupleIndexNum
+)
+
+func encodePartitionInfoToLattice(pi *model.PartitionInfo) Tuple {
+	partitions := make(partitionMap, len(pi.Definitions))
+	for _, d := range pi.Definitions {
+		partitions[d.Name.L] = encodePartitionDefinitionToLattice(d.Clone())
+	}
+	columns := make(Tuple, 0, len(pi.Columns))
+	for _, c := range pi.Columns {
+		columns = append(columns, Singleton(c.L))
+	}
+	num := Singleton(pi.Num)
+	// Only range and list partition num can be changed.
+	if pi.Type == model.PartitionTypeRange || pi.Type == model.PartitionTypeList {
+		num = Uint64(pi.Num)
+	}
+	return Tuple{
+		Singleton(pi.Type),
+		Singleton(pi.Expr),
+		columns,
+		Singleton(pi.Enable),
```

Comment:
by reading the comment of `pi.Enable`, I think this is a TiDB specified attribute so we don't need it.